### PR TITLE
feat(FlowItem): Add slot for header content

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #850

## Summary

Allow FlowItem to pass content directly to Panel's header-content slot.
Users should still be able to use the heading and summary attributes/props to hydrate Panel's header-content. But using the header-content at Flow Item would take precedence over those props.

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco can you review this?

File: src/demos/flow-item/basic.html [hunk 1]
```diff
@@ -15,6 +15,7 @@
       <section class="example-container">
         <h1>calcite-flow-item</h1>
 
+        <h2>Basic</h2>
```
Comment (jcfranco) [inline src/demos/flow-item/basic.html]: Should these two `<h2>`s be merged?

File: src/components/calcite-flow-item/calcite-flow-item.tsx [hunk 2]
```diff
@@ -391,7 +402,7 @@ export class CalciteFlowItem {
           dir={dir}
         >
           {this.renderBackButton(dir === "rtl")}
-          {this.renderHeader()}
+          {this.renderHeaderSlot() || this.renderHeader()}
```
Comment (jcfranco) [inline src/components/calcite-flow-item/calcite-flow-item.tsx]: Can `renderHeader` be used as the header slot's default content?

```tsx
  renderHeaderSlot(): VNode {
    const hasHeaderContentSlot = !!this.el.querySelector(`[slot=${SLOTS.headerContent}]`);

    return hasHeaderContentSlot ? (
      <div class={CSS.header} slot={PANEL_SLOTS.headerContent}>
        <slot name={SLOTS.headerContent} >{this.renderHeader()}</slot>
      </div>
    ) : null;
  }
```

**Comment by jcfranco** [comment]
Also, for consistency, can you use `flow-item` as the commit type's scope (`feat(flow-item): Add slot for header content`)?

**Pull request closed without merging**